libshaderc/include/shaderc.h

```diff
@@ -105,6 +111,11 @@ void shaderc_compile_options_release(shaderc_compile_options_t options);
 void shaderc_compile_options_add_macro_definition(
     shaderc_compile_options_t options, const char* name, const char* value);
 
+// Set flags for the rules on which warnings or errors will be issued. This can
+// either be an individual shaderc_msg_rules flag, or a set of flags or'd together
+void shaderc_compile_options_set_message_rules(
```

Comment:
Two other alternatives here:

1) add an "int version" parameter now to the target env function and document it as unused for now; that at least gets the API in place and avoids having to change it in the future.

2) down the road, add a second set_target_env variant which has the "int version" parameter; document the original set_target_env version to use the highest supported version.

Either of those would avoid having separate target & version functions and the "when to validate" issue. I don't have a strong preference offhand.